### DISCUSSION TITLE
Is there a way to run a method when accessing any page of the resource?

### DISCUSSION THREAD

**Discussion by diogogpinto** [filamentphp/filament#10332] [Help] [answered]
I want to run a function anytime a page is visited in the resource. Something like:

```php
        activity()
            ->performedOn(static::getRecord())
            ->event('retrieved')
            ->log(__('Viewed Record'));
```

Currently I'm using this on every page, in the rendered() method. I would like to know if there's any way to perform this code on a resource level, to every page registered in getPages()

Thank you!

**Comment by leandrocfe** [comment, marked as answer]
`getRecord()` is available when the record is already created, so you couldn't use it on all pages. However, you could create a trait and reuse it in the ViewPage and EditPage, for example.

**Reply by diogogpinto** [reply]
Hello @leandrocfe 

Thank you! That's what I did. I've created the trait and use it in every sub page.

**Reply by leandrocfe** [reply]
If it works, mark as answered please
